hw/top_earlgrey/doc/design/top_generation.md

```diff
@@ -0,0 +1,83 @@
+# Earl Grey Top Generation
+
+Earl Grey is assembled/generated by our in-house integration tool [topgen](../../../../util/topgen/README.md).
+
+## Autogenerated Files warning
+
+The topgen tool autogenerates a number of modules and other files when it runs.
+A number of these files have been committed into the repository for browsing
+purposes. Please do not modify these files manually:
+- `rtl/top_earlgrey.sv`
+- The crossbar modules
+- The interrupt controller
+
+## How a top module is created
+
+The top module `rtl/top_earlgrey.sv` is created by `topgen` using the following command:
+
+```bash
+../../util/topgen.py -t data/top_earlgrey.hjson -o . -v
+```
+
+The data declared in `data/top-earlgrey.hjson` is the single-source-of-truth
+for the autogeneration of collateral for the Earl Grey design.
```

Comment:
I really don't like how we use autogeneration to describe this. We generate form templates, not "by oneself", but we cannot change it in a single place when the word is used so pervasively.